addons/sale_crm/models/sale_order.py

```diff
@@ -13,3 +13,15 @@ class SaleOrder(models.Model):
 
     def action_confirm(self):
         return super(SaleOrder, self.with_context({k:v for k,v in self._context.items() if k != 'default_tag_ids'})).action_confirm()
+
+    def message_post(self, **kwargs):
```

Comment:
Hello @adsh-odoo Can't we handle this with compute method?(Make expected_revenue computed) Can you give It a try? The main drawback here is message_post will be called even If the user will log in the quotation. And this whole code will still execute even If we don't need to.
Hello @jeh-odoo What is your thoughts on this?